Sources/Commands/SwiftPackageTool.swift

```diff
@@ -0,0 +1,295 @@
+        
+                let projectName: String
+                let dstdir: String
+                let packageName = rootPackage.name
+        
+                switch outpath {
+                case let outpath? where outpath.hasSuffix(".xcodeproj"):
+                    // if user specified path ending with .xcodeproj, use that
+                    projectName = String(outpath.basename.characters.dropLast(10))
+                    dstdir = outpath.parentDirectory
+                case let outpath?:
+                    dstdir = outpath
+                    projectName = packageName
+                case _:
+                    dstdir = opts.path.root
+                    projectName = packageName
+                }
+                let outpath = try Xcodeproj.generate(dstdir: dstdir.abspath, projectName: projectName, srcroot: opts.path.root, modules: xcodeModules, externalModules: externalXcodeModules, products: products, options: opts)
+        
+                print("generated:", outpath.prettyPath)
+                
+            case .DumpPackage(let packagePath):
+                
+                let root = packagePath ?? opts.path.root
+                let manifest = try parseManifest(path: root, baseURL: root)
+                let package = manifest.package
+                let json = try jsonString(package: package)
+                print(json)
+            }
+        
+        } catch {
+            handle(error: error, usage: usage)
+        }
+    }
+
+    private func usage(_ print: (String) -> Void = { print($0) }) {
+        //     .........10.........20.........30.........40.........50.........60.........70..
+        print("OVERVIEW: Perform operations on a swift package")
+        print("")
+        print("USAGE: swift package [command] [options]")
+        print("")
+        print("COMMANDS:")
+        print("  init[=<type>]                 Initialize a new package (executable|library)")
+        print("  fetch                         Fetch package dependencies")
+        print("  update                        Update package dependencies")
+        print("  generate-xcodeproj[=<path>]   Generates an Xcode project")
+        print("  show-dependencies[=<format>]  Print dependency graph (text|dot|json)")
+        print("  dump-package[=<path>]         Print Package.swift as JSON")
+        print("")
+        print("OPTIONS:")
+        print("  --chdir <path>       Change working directory before any command [-C]")
+        print("  --color <mode>       Specify color mode (auto|always|never)")
+        print("  --verbose            Increase verbosity of informational output [-v]")
+        print("  -Xcc <flag>          Pass flag through to all C compiler instantiations")
+        print("  -Xlinker <flag>      Pass flag through to all linker instantiations")
+        print("  -Xswiftc <flag>      Pass flag through to all Swift compiler instantiations")
```

Comment:
Random thought, I wonder if `xcodeproj` generation might be ok to leave on `swift build`? That would allow us to keep all build related options there. From a certain perspective it is a "different way of building" although obvious from other perspectives it is a completely separate command.